test/k23si/QueryTest.cpp

```diff
@@ -624,6 +625,61 @@ seastar::future<> runScenario06() {
     });
 }
 
+// Read an erased record via query
+seastar::future<> runScenario07() {
+    K2LOG_I(log::k23si, "runScenario07");
+
+    K2LOG_I(log::k23si, "Read an erased record via query");
+    k2::K2TxnOptions options{};
+    options.syncFinalize = true;
+    return _client.beginTxn(options)
+    .then([this] (k2::K2TxnHandle&& t) {
+        writeTxn = std::move(t);
+        return seastar::make_ready_future<>();
+    })
+    .then([this] () {
+        return _client.getSchema(collname, "schema", 1);
+    })
+    .then([this] (auto&& response) {
+        auto& [status, schemaPtr] = response;
+        K2EXPECT(log::k23si, status.is2xxOK(), true);
+
+        k2::dto::SKVRecord record(collname, schemaPtr);
+        record.serializeNext<k2::String>("default");
+        record.serializeNext<k2::String>("scenario07");
+        record.serializeNext<k2::String>("");
+        return writeTxn.write<k2::dto::SKVRecord>(record);
+    })
+    .then([this] (auto&& response) {
+        K2EXPECT(log::k23si, response.status, k2::dto::K23SIStatus::Created);
+        return _client.getSchema(collname, "schema", 1);
+    })
+    .then([this] (auto&& response) {
+        auto& [status, schemaPtr] = response;
+        K2EXPECT(log::k23si, status.is2xxOK(), true);
+
+        k2::dto::SKVRecord record(collname, schemaPtr);
+        record.serializeNext<k2::String>("default");
+        record.serializeNext<k2::String>("scenario07");
+        record.serializeNext<k2::String>("");
+        return writeTxn.erase(record);
+    })
+    .then([this] (auto&& response) {
+        K2EXPECT(log::k23si, response.status, k2::dto::K23SIStatus::Created);
+        return seastar::make_ready_future<>();
+    })
+    .then([this] () {
+        return writeTxn.end(true);
+    })
+    .then([] (auto&& response) {
+        K2EXPECT(log::k23si, response.status, k2::dto::K23SIStatus::OK);
+        return seastar::make_ready_future<>();
+    })
+    .then([this] () {
+        return doQuery("scenario07", "scenario07", -1, false, 0, 1).discard_result();
+    });
```

Comment:
Is the point here that you get a 200 response on the query with no records returned? I think you should assert that after doQuery

Reply:
doQuery is a test method that does that already, and it asserts that the number of records returned is as expected, 0 in this case.